From 080c1ac69d99cd4dbf498ff186d238e3c51118eb Mon Sep 17 00:00:00 2001
From: Maulik Manvar <maulik.manvar@softwebsolutions.com>
Date: Tue, 11 Apr 2023 15:15:50 +0530
Subject: [PATCH 1/3] Added peripheral support for sm2s-rzg2ul board

-Added DTS changes to bring up peripheral support like eeprom,
da7212 codec,RTC,Temperature sensor,SD Card, EMMC,CSI,CAN,PWM interfaces,
Display Interfaces.

Signed-off-by: Maulik Manvar <maulik.manvar@softwebsolutions.com>
---
 .../boot/dts/renesas/r9a07g043u11-smarc.dts   |   70 +-
 arch/arm64/boot/dts/renesas/r9a07g043u11.dtsi |   13 +
 .../renesas/sm2s_rzg2ul-smarc-display.dtsi    | 1050 +++++++++++++++++
 .../boot/dts/renesas/sm2s_rzg2ul-smarc.dtsi   |  829 +++++++++++++
 4 files changed, 1955 insertions(+), 7 deletions(-)
 create mode 100755 arch/arm64/boot/dts/renesas/r9a07g043u11.dtsi
 create mode 100644 arch/arm64/boot/dts/renesas/sm2s_rzg2ul-smarc-display.dtsi
 create mode 100644 arch/arm64/boot/dts/renesas/sm2s_rzg2ul-smarc.dtsi

diff --git a/arch/arm64/boot/dts/renesas/r9a07g043u11-smarc.dts b/arch/arm64/boot/dts/renesas/r9a07g043u11-smarc.dts
index 121e55282d18..a7c2ccd54633 100644
--- a/arch/arm64/boot/dts/renesas/r9a07g043u11-smarc.dts
+++ b/arch/arm64/boot/dts/renesas/r9a07g043u11-smarc.dts
@@ -1,15 +1,71 @@
-// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+// SPDX-License-Identifier: GPL-2.0
 /*
- * Device Tree Source for the RZ/G2UL Type-1 SMARC EVK board
+ * Device Tree Source for the RZG2UL Single A55 Evaluation board
  *
  * Copyright (C) 2022 Renesas Electronics Corp.
  */
 
 /dts-v1/;
-#include "r9a07g043.dtsi"
-#include "rzg2ul-smarc.dtsi"
-
+#include "r9a07g043u11.dtsi"
+//#include "rzg2ul-smarc.dtsi"
+//#include "sm2s_rzg2ul-smarc.dtsi"
+#include "sm2s_rzg2ul-smarc-display.dtsi"
 / {
-	model = "Renesas SMARC EVK based on r9a07g043u11";
-	compatible = "renesas,smarc-evk", "renesas,r9a07g043u11", "renesas,r9a07g043";
+	model = "Evaluation board based on r9a07g043u11";
+	compatible = "renesas,r9a07g043u11-smarc",
+		     "renesas,r9a07g043",
+		     "renesas,r9a07g043u11";
+
+	memory@48000000 {
+		device_type = "memory";
+		/* first 128MB is reserved for secure area. */
+		reg = <0x0 0x48000000 0x0 0x38000000>;
+	};
+
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		global_cma: linux,cma@58000000 {
+			compatible = "shared-dma-pool";
+			linux,cma-default;
+			reusable;
+			reg = <0x0 0x58000000 0x0 0x10000000>;
+		};
+		mmp_reserved: linux,multimedia {
+			compatible = "shared-dma-pool";
+			reusable;
+			reg = <0x00000000 0x68000000 0x0 0x8000000>;
+		};
+	};
+
+	mmngr {
+		compatible = "renesas,mmngr";
+		memory-region = <&mmp_reserved>;
+	};
+
+	mmngrbuf {
+		compatible = "renesas,mmngrbuf";
+	};
+
+	vspm_if {
+		compatible = "renesas,vspm_if";
+	};
+};
+
+&ehci0 {
+	memory-region = <&global_cma>;
+};
+
+&ohci0 {
+	memory-region = <&global_cma>;
+};
+
+&ehci1 {
+	memory-region = <&global_cma>;
+};
+
+&ohci1 {
+	memory-region = <&global_cma>;
 };
diff --git a/arch/arm64/boot/dts/renesas/r9a07g043u11.dtsi b/arch/arm64/boot/dts/renesas/r9a07g043u11.dtsi
new file mode 100755
index 000000000000..0a8c3cd505d1
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/r9a07g043u11.dtsi
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/*
+ * Device Tree Source for the RZ/G2UL R9A07G043U11 SoC specific parts
+ *
+ * Copyright (C) 2022 Renesas Electronics Corp.
+ */
+
+/dts-v1/;
+#include "r9a07g043.dtsi"
+
+/ {
+	compatible = "renesas,r9a07g043ul1", "renesas,r9a07g043";
+};
diff --git a/arch/arm64/boot/dts/renesas/sm2s_rzg2ul-smarc-display.dtsi b/arch/arm64/boot/dts/renesas/sm2s_rzg2ul-smarc-display.dtsi
new file mode 100644
index 000000000000..8d2199f1e3b1
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/sm2s_rzg2ul-smarc-display.dtsi
@@ -0,0 +1,1050 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Device Tree Source for the RZG2UL Evaluation Board Kit
+ *
+ * Copyright (C) 2022 Renesas Electronics Corp.
+ */
+
+#include <dt-bindings/pinctrl/rzg2l-pinctrl.h>
+#include <dt-bindings/gpio/gpio.h>
+
+/*
+ * If Display is enabled then SW_DISP_EN should be 1 otherwise 0.
+ * Please change below MACROs corespondingly to SW1 setting
+ */
+
+#define	SW_DISP_EN	1
+
+/ {
+	aliases {
+		i2c0 = &i2c0;
+		i2c1 = &i2c1;
+		i2c2 = &i2c2;
+		serial0 = &scif0;
+		ethernet0 = &eth0;
+	};
+
+	chosen {
+		bootargs = "ignore_loglevel";
+		stdout-path = "serial0:115200n8";
+	};
+
+	audio_mclock: audio_mclock {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency = <11289600>;
+	};
+
+        vmain_pd: regulator3 {
+                compatible = "regulator-fixed";
+                regulator-name = "vmain_pd";
+                regulator-min-microvolt = <5000000>;
+                regulator-max-microvolt = <5000000>;
+                regulator-always-on;
+                regulator-boot-on;
+        };
+
+        vcc_5v0: regulator2 {
+                compatible = "regulator-fixed";
+                regulator-name = "vcc_5v0";
+                regulator-min-microvolt = <5000000>;
+                regulator-max-microvolt = <5000000>;
+                vin-supply = <&vmain_pd>;
+                regulator-always-on;
+                regulator-boot-on;
+        };
+
+
+	reg_1p8v: regulator0 {
+		compatible = "regulator-fixed";
+
+		regulator-name = "fixed-1.8V";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+		regulator-boot-on;
+		regulator-always-on;
+	};
+
+	reg_3p3v: regulator1 {
+		compatible = "regulator-fixed";
+
+		regulator-name = "fixed-3.3V";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		regulator-boot-on;
+		regulator-always-on;
+	};
+
+	vbus0_usb2: regulator-vbus0-usb2 {
+		compatible = "regulator-fixed";
+
+		regulator-name = "USB20_VBUS0";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+	};
+
+	snd_rzg2l: sound {
+		compatible = "simple-audio-card";
+                simple-audio-card,name = "sm2s-rzg2ul-card1";
+		simple-audio-card,format = "i2s";
+		simple-audio-card,bitclock-master = <&cpu_dai>;
+		simple-audio-card,frame-master = <&cpu_dai>;
+		simple-audio-card,mclk-fs = <256>;
+
+                simple-audio-card,widgets =
+                        "Headphone",    "Headphone Jack",
+                        "Line", "Line In",
+                        "Microphone",   "Microphone Jack";
+
+                simple-audio-card,audio-routing = "Mic1", "Mic Bias 1",
+                                                  "MIC1", "Mic 1",
+                                                  "Headphone Jack", "HPL",
+                                                  "Headphone Jack", "HPR",
+                                                  "Speaker", "LINE",
+                                                  "Ext Spk", "Line In",
+                                                  "Line", "Mic Bias";
+
+		cpu_dai: simple-audio-card,cpu {
+			sound-dai = <&ssi3>;
+		};
+
+		codec_dai: simple-audio-card,codec {
+			sound-dai = <&da7213_codec>;
+		};
+	};
+
+        x1_clk: x1-clock {
+                compatible = "fixed-clock";
+                #clock-cells = <0>;
+                clock-frequency = <24000000>;
+        };
+
+
+        clk_ext_audio_codec: clock-codec {
+                compatible = "fixed-clock";
+                #clock-cells = <0>;
+                clock-frequency = <12288000>;
+        };
+
+
+        vdd_sd0: regulator-1 {
+                compatible = "regulator-fixed";
+                regulator-name = "vdd_sd0";
+                regulator-min-microvolt = <3300000>;
+                regulator-max-microvolt = <3300000>;
+                regulator-boot-on;
+                enable-active-high;
+                vin-supply = <&reg_3p3v>;
+        };
+
+	vccq_sdhi0: regulator-vccq-sdhi0 {
+		    compatible = "regulator-gpio";
+		    regulator-name = "SDHI0 VccQ";
+		    regulator-min-microvolt = <1800000>;
+		    regulator-max-microvolt = <3300000>;
+		    vin-supply = <&ldo2>;
+                    startup-delay-us = <100>;
+                    u-boot,off-on-delay-us = <12000>;
+                    gpios-states = <0>;
+                    states = <3300000 1
+                          1800000 0>;
+	};
+
+        vccq_sdhi1: regulator-vccq-sdhi1 {
+                compatible = "regulator-gpio";
+
+                regulator-name = "SDHI1 VccQ";
+                regulator-min-microvolt = <1800000>;
+                regulator-max-microvolt = <3300000>;
+
+                gpios-states = <1>;
+                states = <3300000 1
+                          1800000 0>;
+        };
+	
+	vring-ctl0@43000000 {
+		compatible = "vring_uio";
+		reg = <0x0 0x43000000 0x0 0x100000>;
+		no-map;
+	};
+
+	vring-ctl1@43100000 {
+		compatible = "vring_uio";
+		reg = <0x0 0x43100000 0x0 0x100000>;
+		no-map;
+	};
+
+	vring-shm0@43200000 {
+		compatible = "shm_uio";
+		reg = <0x0 0x43200000 0x0 0x300000>;
+		no-map;
+	};
+
+	vring-shm1@43500000 {
+		compatible = "shm_uio";
+		reg = <0x0 0x43500000 0x0 0x300000>;
+		no-map;
+	};
+
+	rsctbl@42f00000 {
+		compatible = "shm_uio";
+		reg = <0x0 0x42f00000 0x0 0x1000>;
+		no-map;
+	};
+
+	mhu-shm@42f01000 {
+		compatible = "shm_uio";
+		reg = <0x0 0x42f01000 0x0 0x1000>;
+		no-map;
+	};
+
+	mbox-uio@10400000 {
+		compatible = "mbox_uio";
+		reg = <0x0 0x10400000 0x0 0x1000>;
+		interrupt-parent = <&gic>;
+		interrupts = <GIC_SPI 75 IRQ_TYPE_LEVEL_HIGH>;
+		status = "okay";
+	};
+
+	x1_clk: x1-clock {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency = <24000000>;
+	};
+
+	clk_ext_camera: clk_ext_camera {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency = <12000000>;
+	};
+
+	ov5645_vdddo_1v8: 1p8v {
+		compatible = "regulator-fixed";
+		regulator-name = "camera_vdddo";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+		regulator-always-on;
+	};
+
+	ov5645_vdda_2v8: 2p8v {
+		compatible = "regulator-fixed";
+		regulator-name = "camera_vdda";
+		regulator-min-microvolt = <2800000>;
+		regulator-max-microvolt = <2800000>;
+		regulator-always-on;
+	};
+
+	ov5645_vddd_1v5: 1p5v {
+		compatible = "regulator-fixed";
+		regulator-name = "camera_vddd";
+		regulator-min-microvolt = <1500000>;
+		regulator-max-microvolt = <1500000>;
+		regulator-always-on;
+	};
+
+	panel {
+                compatible = "panel-lvds";
+                width-mm = <154>;
+                height-mm = <93>;
+                label = "WVGA (15:9)";
+		data-mapping = "jeida-24";
+                status = "okay";
+
+                panel-timing {
+                        clock-frequency = <25505280>;
+                        hactive = <800>;
+                        vactive = <480>;
+                        hback-porch = <30>;
+                        hfront-porch = <30>;
+                        vback-porch = <4>;
+                        vfront-porch = <4>;
+                        hsync-len = <4>;
+                        vsync-len = <4>;
+                };
+
+                port {
+                        lcd_in0: endpoint {
+                                remote-endpoint = <&du_out_rgb>;
+                        };
+                };
+        };
+};
+
+&audio_clk1{
+	clock-frequency = <11289600>;
+};
+
+&audio_clk2{
+	clock-frequency = <12288000>;
+};
+
+&canfd {
+	pinctrl-0 = <&canfd0_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+	channel0 {
+		status = "okay";
+	};
+};
+
+#if (SW_DISP_EN)
+
+&du {
+	pinctrl-0 = <&du_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+
+	ports {
+		port@0 {
+	                reg = <0>;
+			du_out_rgb: endpoint {
+	                        remote-endpoint = <&lcd_in0>;
+			};
+		};
+
+	};
+};
+
+#else
+	&ssi0 {
+		pinctrl-0 = <&ssi0_pins>;
+		pinctrl-names = "default";
+		status = "okay";
+	};
+	
+	&canfd {
+		pinctrl-0 = <&canfd1_pins>;
+		pinctrl-names = "default";
+		status = "okay";
+		channel1 {
+			status = "okay";
+		};
+	};
+#endif
+
+&ehci0 {
+	dr_mode = "otg";
+	status = "okay";
+};
+
+&ehci1 {
+	status = "okay";
+};
+
+&eth0 {
+	pinctrl-0 = <&ether0_pins>;
+	pinctrl-names = "default";
+	phy-handle = <&phy0>;
+	phy-mode = "rgmii-rxid";
+	status = "okay";
+
+	phy0: ethernet-phy@0 {
+		reg = <0>;
+                ti,rx-internal-delay = <0x8>;
+                ti,fifo-depth = <0x01>;
+                ti,min-output-impedance;
+	};
+};
+
+&hsusb {
+	dr_mode = "otg";
+	status = "okay";
+};
+
+&i2c0 {
+	pinctrl-0 = <&i2c0_pins>;
+	pinctrl-names = "default";
+        clock-frequency = <400000>;
+	status = "okay";
+
+       module_eeprom@50 {
+                compatible = "atmel,24c64";
+                reg = <0x50>;
+                pagesize = <32>;
+        };
+
+        module_eeprom@57 {
+                compatible = "atmel,24c64";
+                reg = <0x57>;
+                pagesize = <32>;
+        };
+
+        da7213_codec: da7212@1a {
+                #sound-dai-cells = <0>;
+                compatible = "dlg,da7213";
+                reg = <0x1a>;
+
+                dlg,micbias1-lvl = <3000>;
+                dlg,micbias2-lvl = <3000>;
+
+                VDDA-supply = <&reg_1p8v>;
+                VDDSP-supply = <&vcc_5v0>;
+                VDDIO-supply = <&reg_1p8v>;
+                VDDMIC-supply= <&reg_3p3v>;
+                clocks = <&clk_ext_audio_codec>;
+                clock-names = "mclk";
+        };
+};
+
+&i2c1 {
+	pinctrl-0 = <&i2c1_pins>;
+	pinctrl-names = "default";
+        clock-frequency = <400000>;
+
+	status = "okay";
+
+       module_eeprom@57 {
+                compatible = "atmel,24c64";
+                reg = <0x57>;
+                pagesize = <32>;
+       };
+};
+
+&i2c2 {
+        #address-cells = <1>;
+        #size-cells = <0>;
+        pinctrl-names = "default";
+        pinctrl-0 = <&i2c2_pins>;
+        clock-frequency = <400000>;
+        status="okay";
+
+
+        pmic1: pmic@30 {
+                compatible = "ricoh,rn5t567";
+                reg = <0x30>;
+                pmic-id = <0>;
+                sleep-sequence = /bits/ 8 <
+                                0x16    0x2b
+                                0x17    0x49
+                                0x1b    0x2b
+                                0x1c    0x67
+                                0x1f    0x0b
+                                0x32    0x03
+                                0x30    0x03
+                        >;
+                
+
+                regulators {
+                        dcdc1:DCDC1 {
+                                regulator-name = "VCC_DRAM_VPU_0V9";
+                                regulator-always-on;
+                                regulator-min-microvolt = <900000>;
+                                regulator-max-microvolt = <1200000>;
+                        };
+                        dcdc2:DCDC2 {
+                                regulator-name = "VCC_ARM_0V9";
+                                regulator-always-on;
+                                regulator-min-microvolt = <900000>;
+                                regulator-max-microvolt = <3300000>;
+                        };
+                        ldo1:LDO1 {
+                                regulator-name = "VCC_PHY_0V9";
+                                regulator-always-on;
+                                regulator-min-microvolt = <900000>;
+                                regulator-max-microvolt = <1800000>;
+                        };
+                        ldo2:LDO2 {
+                                regulator-name = "VCC_LDO12_1V2";
+                                regulator-always-on;
+                                regulator-min-microvolt = <1000000>;
+                                regulator-max-microvolt = <1800000>;
+                        };
+		};
+	};
+
+        exp2: gpio@22 {
+                compatible = "ti,tca6424";
+                reg = <0x22>;
+                gpio-controller;
+                #gpio-cells = <2>;
+                interrupt-parent = <&pinctrl>;
+                /*interrupts = <23 IRQ_TYPE_EDGE_FALLING>;
+                interrupt-controller;
+                #interrupt-cells = <2>;
+
+                pinctrl-names = "default";
+                pinctrl-0 = <&main_gpio1_ioexp_intr_pins_default>;*/
+        };
+
+        exp1: gpio@23 {
+                compatible = "ti,tca6424";
+                reg = <0x23>;
+                gpio-controller;
+                #gpio-cells = <2>;
+                interrupt-parent = <&pinctrl>;
+                /*interrupts = <23 IRQ_TYPE_EDGE_FALLING>;
+                interrupt-controller;
+                #interrupt-cells = <2>;
+
+                pinctrl-names = "default";
+                pinctrl-0 = <&main_gpio1_ioexp_intr_pins_default>;*/
+        };
+
+        versa3: versa3@68 {
+                compatible = "renesas,5p35023";
+                reg = <0x68>;
+                #clock-cells = <1>;
+                clocks = <&x1_clk>;
+                clock-names = "x1";
+                assigned-clocks = <&versa3 0>,
+                                   <&versa3 1>,
+                                   <&versa3 2>,
+                                   <&versa3 3>,
+                                   <&versa3 4>,
+                                   <&versa3 5>;
+                assigned-clock-rates =  <12288000>, <25000000>,
+                                        <12000000>, <11289600>,
+                                        <11289600>, <24000000>;
+                clock-divider-read-only = <1>, <1>, <1>, <1>, <1>;
+                clock-flags = <2176>, <2176>, <2176>, <2052>, <2176>, <2048>;
+        };
+
+        tmp103: tmp103@75 {
+                compatible = "ti,tmp103";
+                reg = <0x75>;
+        };
+
+        sys_rtc: rtc@32 {
+                compatible = "microcrystal,rv8803";
+                reg = <0x32>;
+        };
+
+};
+
+&i2c3 {
+	clock-frequency = <100000>;
+	pinctrl-0 = <&i2c3_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+
+	ov5640: ov5640@3c {
+		compatible = "ovti,ov5640";
+		reg = <0x3c>;
+		clock-names = "xclk";
+		clocks = <&clk_ext_camera>;
+		clock-frequency = <12000000>;
+		DOVDD-supply = <&ov5645_vdddo_1v8>;
+		AVDD-supply = <&ov5645_vdda_2v8>;
+		DVDD-supply = <&ov5645_vddd_1v5>;
+		/*vdddo-supply = <&ov5645_vdddo_1v8>;
+		vdda-supply = <&ov5645_vdda_2v8>;
+		vddd-supply = <&ov5645_vddd_1v5>;*/
+
+		enable-gpios = <&exp1 1 GPIO_ACTIVE_LOW>;
+		reset-gpios = <&exp1 3 GPIO_ACTIVE_HIGH>;
+
+		port {
+			ov5645_to_csi: endpoint {
+		       		clock-lanes = <0>;
+		       		data-lanes = <1 2>;
+		       		remote-endpoint = <&csi2_in>;
+                 		bus-width = <8>;
+		                hsync-active = <0>;
+                		vsync-active = <1>;
+		                pclk-sample = <0>;
+	       		};
+		};
+	};
+};
+
+&ssi3 {
+	    pinctrl-0 = <&ssi3_pins>;
+	    pinctrl-names = "default";
+	    status = "okay";
+};
+
+&usb2_phy0 {
+	pinctrl-0 = <&usb0_pins>;
+	pinctrl-names = "default";
+	
+	//Enable VBUS for USB0
+	gpio = <&exp2 15 GPIO_ACTIVE_HIGH>;	
+	
+	vbus-supply = <&vbus0_usb2>;
+	status = "okay";
+};
+
+&usb2_phy1 {
+	/*pinctrl-0 = <&usb1_pins>;
+	pinctrl-names = "default";*/
+
+	status = "okay";
+};
+
+
+&spi0 {
+	pinctrl-0 = <&spi0_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+	/*flash@0 {
+		compatible="winbond,w25q64dw","jedec,spi-nor";
+		reg = <0>;
+		m25p,fast-read;
+		spi-max-frequency = <50000000>;
+		spi-tx-bus-width = <1>;
+		spi-rx-bus-width = <1>;
+		partitions {
+			compatible="fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+			partition@000000 {
+				label = "spi0_bl2";
+				reg = <0x00000000 0x0001D200>;
+				read-only;
+			};
+			partition@01D200 {
+				label = "spi0_fip";
+				reg = <0x0001D200 0x001C2E00>;
+				read-only;
+			};
+			partition@1E0000 {
+				label = "spi0_env";
+				reg = <0x001E0000 0x00020000>;
+				read-only;
+			};
+			partition@200000 {
+				label = "spi0_test-area";
+				reg = <0x00200000 0x00E00000>;
+			};
+		};
+	};*/
+
+	spidev@0x00 {
+		compatible="winbond,w25q64dw","jedec,spi-nor";
+		spi-max-frequency = <100000>;
+		reg = <0>;
+	};
+};
+
+
+&spi2 {
+        pinctrl-0 = <&spi2_pins>;
+        pinctrl-names = "default";
+        status = "okay";
+        /*flash@0 {
+                compatible="winbond,w25q64dw","jedec,spi-nor";
+                reg = <0>;
+                m25p,fast-read;
+                spi-max-frequency = <50000000>;
+                spi-tx-bus-width = <1>;
+                spi-rx-bus-width = <1>;
+                partitions {
+                        compatible="fixed-partitions";
+                        #address-cells = <1>;
+                        #size-cells = <1>;
+                        partition@000000 {
+                                label = "spi0_bl2";
+                                reg = <0x00000000 0x0001D200>;
+                                read-only;
+                        };
+                        partition@01D200 {
+                                label = "spi0_fip";
+                                reg = <0x0001D200 0x001C2E00>;
+                                read-only;
+                        };
+                        partition@1E0000 {
+                                label = "spi0_env";
+                                reg = <0x001E0000 0x00020000>;
+                                read-only;
+                        };
+                        partition@200000 {
+                                label = "spi0_test-area";
+                                reg = <0x00200000 0x00E00000>;
+                        };
+                };
+        };*/
+
+        spidev@0x00 {
+                compatible="winbond,w25q64dw","jedec,spi-nor";
+                spi-max-frequency = <100000>;
+                reg = <0>;
+        };
+
+
+};
+
+&scif0 {
+	pinctrl-0 = <&scif0_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+};
+	
+
+&scif1 {
+    pinctrl-0 = <&scif1_pins>;
+    pinctrl-names = "default";
+    status = "okay";
+};
+
+&scif2 {
+    pinctrl-0 = <&scif2_pins>;
+    pinctrl-names = "default";
+    status = "okay";
+};
+
+
+&scif4 {
+    pinctrl-0 = <&scif4_pins>;
+    pinctrl-names = "default";
+    status = "okay";
+};
+
+//Comments:sci0 : M3 and M6 pins are used but Name is not verified with r9...It is serial.
+
+&sci0 {
+    pinctrl-0 = <&sci0_pins>;
+    pinctrl-names = "default";
+    status = "okay";
+};
+	
+&phyrst {
+	status = "okay";
+};
+
+&ohci0 {
+	dr_mode = "otg";
+	status = "okay";
+};
+
+&ohci1 {
+	status = "okay";
+};
+
+&pinctrl {
+	pinctrl-0 = <&sound_clk_pins>;
+	pinctrl-names = "default";
+
+
+	i2c2_pins: i2c2 {
+               pinmux = <RZG2L_PORT_PINMUX(0, 3, 4)>, /* RIIC2_SCL */
+                         <RZG2L_PORT_PINMUX(0, 2, 4)>; /*RIIC2_SDA*/
+	};
+	
+	i2c3_pins: i2c3 {
+               pinmux = <RZG2L_PORT_PINMUX(0, 1, 3)>, /* RIIC3_SCL */
+                         <RZG2L_PORT_PINMUX(0, 0, 3)>; /*RIIC3_SDA*/
+	};
+
+	canfd0_pins: can0 {
+		pinmux = <RZG2L_PORT_PINMUX(6, 1, 3)>, /* TX */
+			 <RZG2L_PORT_PINMUX(6, 2, 3)>; /* RX */
+	};
+
+	ether0_pins: eth0 {
+		pinmux = 
+			 <RZG2L_PORT_PINMUX(4, 3, 1)>, /* ET0_MDC */
+			 <RZG2L_PORT_PINMUX(4, 4, 1)>, /* ET0_MDIO */
+			 <RZG2L_PORT_PINMUX(1, 0, 1)>, /* ET0_TXC */
+			 <RZG2L_PORT_PINMUX(1, 1, 1)>, /* ET0_TX_CTL */
+			 <RZG2L_PORT_PINMUX(1, 2, 1)>, /* ET0_TXD0 */
+			 <RZG2L_PORT_PINMUX(1, 3, 1)>, /* ET0_TXD1 */
+			 <RZG2L_PORT_PINMUX(1, 4, 1)>, /* ET0_TXD2 */
+			 <RZG2L_PORT_PINMUX(2, 0, 1)>, /* ET0_TXD3 */
+			 <RZG2L_PORT_PINMUX(3, 0, 1)>, /* ET0_RXC */
+			 <RZG2L_PORT_PINMUX(3, 1, 1)>, /* ET0_RX_CTL */
+			 <RZG2L_PORT_PINMUX(3, 2, 1)>, /* ET0_RXD0 */
+			 <RZG2L_PORT_PINMUX(3, 3, 1)>, /* ET0_RXD1 */
+			 <RZG2L_PORT_PINMUX(4, 0, 1)>, /* ET0_RXD2 */
+			 <RZG2L_PORT_PINMUX(4, 1, 1)>; /* ET0_RXD3 */
+	};
+
+	i2c0_pins: i2c0 {
+		pins = "RIIC0_SDA", "RIIC0_SCL";
+		input-enable;
+	};
+
+	i2c1_pins: i2c1 {
+		pins = "RIIC1_SDA", "RIIC1_SCL";
+		input-enable;
+	};
+
+
+
+/* Support pinctrl for uSD function of SDHI0*/
+//For SD Card
+	sdhi0_pins: sd0 {
+		sd0_data {
+			pins =  "SD0_DATA0", "SD0_DATA1", "SD0_DATA2",
+				"SD0_DATA3";
+			power-source  = <1800>;
+		};
+
+		sd0_ctrl {
+			pins = "SD0_CLK", "SD0_CMD";
+			power-source = <1800>;
+		};
+	};
+	
+// For MMC
+	sdhi1_pins: sd1 {
+		sd1_data {
+			pins =	"SD1_DATA0", "SD1_DATA1",
+				"SD1_DATA2", "SD1_DATA3";
+			power-source  = <1800>;
+		};
+
+		sd1_ctrl {
+			pins = "SD1_CLK", "SD1_CMD";
+			power-source  = <1800>;
+		};
+	};
+	
+	
+#if (SW_DISP_EN)
+
+	du_pins: du {
+		data {
+
+			pinmux = <RZG2L_PORT_PINMUX(13, 0, 6)>,
+				 <RZG2L_PORT_PINMUX(13, 4, 6)>,
+				 <RZG2L_PORT_PINMUX(13, 3, 6)>,
+				 <RZG2L_PORT_PINMUX(12, 1, 6)>,
+				 <RZG2L_PORT_PINMUX(13, 2, 6)>,
+				 <RZG2L_PORT_PINMUX(14, 0, 6)>,
+				 <RZG2L_PORT_PINMUX(16, 0, 6)>,
+				 <RZG2L_PORT_PINMUX(15, 0, 6)>,
+				 <RZG2L_PORT_PINMUX(16, 1, 6)>,
+				 <RZG2L_PORT_PINMUX(15, 1, 6)>,
+				 <RZG2L_PORT_PINMUX(15, 3, 6)>,
+				 <RZG2L_PORT_PINMUX(18, 0, 6)>,
+				 <RZG2L_PORT_PINMUX(17, 2, 6)>,
+				 <RZG2L_PORT_PINMUX(17, 1, 6)>,
+				 <RZG2L_PORT_PINMUX(18, 1, 6)>,
+				 <RZG2L_PORT_PINMUX(18, 2, 6)>,
+				 <RZG2L_PORT_PINMUX(17, 3, 6)>,
+				 <RZG2L_PORT_PINMUX(18, 3, 6)>;
+			 drive-strength = <2>;
+		};
+
+		sync {
+			pinmux = <RZG2L_PORT_PINMUX(11, 0, 6)>, /* HSYNC */
+				 <RZG2L_PORT_PINMUX(12, 0, 6)>; /* VSYNC */
+	                  	 drive-strength = <2>;
+		};
+
+		de {
+			pinmux = <RZG2L_PORT_PINMUX(11, 1, 6)>; /* DE */
+				 drive-strength = <2>;
+		};
+		
+		clk {
+			pinmux = <RZG2L_PORT_PINMUX(11, 3, 6)>; /* CLK */
+		};
+	};
+#else
+	ssi0_pins: ssi0 {
+		pinmux = <RZG2L_PORT_PINMUX(11, 0, 1)>, /* SSI0_BCK */
+			 <RZG2L_PORT_PINMUX(11, 1, 1)>, /* SSI0_RCK */
+			 <RZG2L_PORT_PINMUX(11, 2, 1)>, /* SSI0_TXD */
+			 <RZG2L_PORT_PINMUX(11, 3, 1)>; /*SSI0_RXD*/
+
+	};
+	
+	canfd1_pins: can1 {
+		pinmux = <RZG2L_PORT_PINMUX(14, 0, 3)>, 
+			 <RZG2L_PORT_PINMUX(14, 1, 3)>; 
+	};
+
+#endif
+	
+	ssi3_pins: ssi3 {
+		       pinmux = <RZG2L_PORT_PINMUX(10, 1, 3)>, /* SSI3_BCK */
+		                 <RZG2L_PORT_PINMUX(10, 2, 3)>, /*SSI3_RCK*/
+		                 <RZG2L_PORT_PINMUX(10, 3, 3)>, /*SSI3_TXD*/
+		                 <RZG2L_PORT_PINMUX(10, 4, 3)>; /*SSI3_RXD*/
+	};
+
+
+	sound_clk_pins: sound_clk {
+		pins = "AUDIO_CLK1", "AUDIO_CLK2";
+		input-enable;
+	};
+
+
+	usb0_pins: usb0 {
+		pinmux = /*<RZG2L_PORT_PINMUX(5, 0, 1)>,
+			 <RZG2L_PORT_PINMUX(5, 2, 1)>,
+			 <RZG2L_PORT_PINMUX(5, 3, 1)>;*/ /* OTG_ID */
+			 <RZG2L_PORT_PINMUX(5, 2, 1)>; /* OVC */
+	};
+
+	usb1_pins: usb1 {
+		pinmux = <RZG2L_PORT_PINMUX(6, 0, 1)>;
+	};
+
+
+
+	spi0_pins: rspi0 {
+		       pinmux = <RZG2L_PORT_PINMUX(9, 0, 2)>, /* RSPI0_CK */
+		                 <RZG2L_PORT_PINMUX(9, 1, 2)>, /*RSPI0_MOSI*/
+		                 <RZG2L_PORT_PINMUX(9, 2, 2)>, /*RSPI0_MISO*/
+		                 <RZG2L_PORT_PINMUX(9, 3, 2)>; /*RSPI0_SSL*/
+	};
+
+
+	spi2_pins: rspi2 {
+		       pinmux = <RZG2L_PORT_PINMUX(7, 0, 3)>, /* RSPI2_CK */
+		                 <RZG2L_PORT_PINMUX(7, 1, 3)>, /*RSPI2_MOSI*/
+		                 <RZG2L_PORT_PINMUX(7, 2, 3)>, /*RSPI2_MISO*/
+		                 <RZG2L_PORT_PINMUX(7, 3, 3)>; /*RSPI2_SSL*/
+	};
+	
+	scif0_pins: scif0 {
+			pinmux = <RZG2L_PORT_PINMUX(6, 4, 6)>, /* TxD */
+				 <RZG2L_PORT_PINMUX(6, 3, 6)>; /* RxD */
+	};
+
+
+	scif1_pins: scif1 {
+		       pinmux = <RZG2L_PORT_PINMUX(8, 0, 3)>, /* SCIF1_RXD */
+		                 <RZG2L_PORT_PINMUX(8, 1, 3)>,/*SCIF1_TXD*/
+		                 <RZG2L_PORT_PINMUX(8, 2, 3)>, /*SCIF1_CTS*/
+		                 <RZG2L_PORT_PINMUX(8, 3, 3)>; /*SCIF1_RTS*/
+	};
+
+
+
+	scif2_pins: scif2 {
+		       pinmux = <RZG2L_PORT_PINMUX(5, 0, 2)>, /* SCIF2_TXD */
+		                 <RZG2L_PORT_PINMUX(5, 1, 2)>, /*SCIF2_RXD*/
+		                 <RZG2L_PORT_PINMUX(5, 3, 2)>, /*SCIF2_CTS*/
+		                 <RZG2L_PORT_PINMUX(5, 4, 2)>; /*SCIF2_RTS*/
+	};
+
+
+	scif4_pins: scif4 {
+		       pinmux = <RZG2L_PORT_PINMUX(18, 4, 7)>, /* SCIF4_RXD */
+		                 <RZG2L_PORT_PINMUX(18, 5, 7)>; /*SCIF4_TXD*/
+	};
+	
+	sci0_pins: sci0 {
+               pinmux =  <RZG2L_PORT_PINMUX(2, 2, 5)>, /*SCI0_TXD*/
+                         <RZG2L_PORT_PINMUX(2, 3, 5)>; /* SCI0_RXD*/ 
+	};
+
+	mtu7_pins:mtu7 {
+	               pinmux =  <RZG2L_PORT_PINMUX(6, 0, 5)>;
+	};
+
+};
+
+&sbc {
+	status = "okay";
+
+	flash@0 {
+		compatible = "micron,mt25qu512a", "jedec,spi-nor";
+		reg = <0>;
+		m25p,fast-read;
+		spi-max-frequency = <50000000>;
+		spi-tx-bus-width = <1>;
+		spi-rx-bus-width = <4>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@000000 {
+				label = "bl2";
+				reg = <0x00000000 0x0001D200>;
+				read-only;
+			};
+			partition@01D200 {
+				label = "fip";
+				reg = <0x0001D200 0x001C2E00>;
+				read-only;
+			};
+			partition@1E0000 {
+				label = "env";
+				reg = <0x001E0000 0x00020000>;
+				read-only;
+			};
+			partition@200000 {
+				label = "test-area";
+				reg = <0x00200000 0x00E00000>;
+			};
+		};
+	};
+};
+
+
+
+
+//SD Card
+&sdhi0 {
+	pinctrl-0 = <&sdhi0_pins>;
+	pinctrl-names = "default";
+	vmmc-supply = <&vdd_sd0>;
+	vqmmc-supply = <&vccq_sdhi0>;
+	bus-width = <4>;
+	status = "okay";
+	disable-wp;
+};
+
+//eMMC
+&sdhi1 {
+        pinctrl-0 = <&sdhi1_pins>;
+        pinctrl-1 = <&sdhi1_pins>;
+        pinctrl-names = "default", "state_uhs";
+
+        vmmc-supply = <&reg_3p3v>;
+        vqmmc-supply = <&reg_1p8v>;
+        bus-width = <4>;
+        mmc-hs200-1_8v;
+        non-removable;
+        fixed-emmc-driver-type = <1>;
+        status = "okay";
+};
+
+&extal_clk {
+	clock-frequency = <24000000>;
+};
+
+&cru {
+	status = "okay";
+};
+
+&csi2 {
+	status = "okay";
+
+	ports {
+		port {
+			csi2_in: endpoint {
+				clock-lanes = <0>;
+				data-lanes = <1 2>;
+				remote-endpoint = <&ov5645_to_csi>;
+			};
+		};
+	};
+};
+
+&wdt0 {
+	status = "okay";
+	timeout-sec = <60>;
+};
+
+&wdt2 {
+	status = "okay";
+	timeout-sec = <60>;
+};
+
+&ostm1 {
+	status = "okay";
+};
+
+&ostm2 {
+	status = "disabled";
+};
+
+&mtu3 {
+	pinctrl-0 = <&mtu7_pins>;
+	pinctrl-names = "default";
+	pwm_mode1 = <7 0>;
+	pwm_complementary = <7 0>;
+        #pwm-cells = <2>;
+        clocks = <&cpg CPG_MOD R9A07G043_MTU_X_MCK_MTU3>;
+        clock-names = "fck";
+	/*pwm-params = <MTU_PWM_NORMAL 0 50>;*/
+	status = "okay";
+};
+
diff --git a/arch/arm64/boot/dts/renesas/sm2s_rzg2ul-smarc.dtsi b/arch/arm64/boot/dts/renesas/sm2s_rzg2ul-smarc.dtsi
new file mode 100644
index 000000000000..cbe61e1670f6
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/sm2s_rzg2ul-smarc.dtsi
@@ -0,0 +1,829 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Device Tree Source for the RZG2UL Evaluation Board Kit
+ *
+ * Copyright (C) 2022 Renesas Electronics Corp.
+ */
+
+#include <dt-bindings/pinctrl/rzg2l-pinctrl.h>
+#include <dt-bindings/gpio/gpio.h>
+
+/*
+ * If Display is enabled then SW_DISP_EN should be 1 otherwise 0.
+ * Please change below MACROs corespondingly to SW1 setting
+ */
+
+#define	SW_DISP_EN	0
+
+/ {
+	aliases {
+		i2c0 = &i2c0;
+		i2c1 = &i2c1;
+		i2c2 = &i2c2;
+		serial0 = &scif0;
+		ethernet0 = &eth0;
+	};
+
+	chosen {
+		bootargs = "ignore_loglevel";
+		stdout-path = "serial0:115200n8";
+	};
+
+	hdmi-out {
+		compatible = "hdmi-connector";
+		type = "d";
+
+		port {
+			hdmi_con_out: endpoint {
+				remote-endpoint = <&adv7513_out>;
+			};
+		};
+	};
+
+	audio_mclock: audio_mclock {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency = <11289600>;
+	};
+
+	reg_1p8v: regulator0 {
+		compatible = "regulator-fixed";
+
+		regulator-name = "fixed-1.8V";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+		regulator-boot-on;
+		regulator-always-on;
+	};
+
+	reg_3p3v: regulator1 {
+		compatible = "regulator-fixed";
+
+		regulator-name = "fixed-3.3V";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		regulator-boot-on;
+		regulator-always-on;
+	};
+
+	vbus0_usb2: regulator-vbus0-usb2 {
+		compatible = "regulator-fixed";
+
+		regulator-name = "USB20_VBUS0";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+	};
+
+	snd_rzg2l: sound {
+		compatible = "simple-audio-card";
+		simple-audio-card,format = "i2s";
+		simple-audio-card,bitclock-master = <&cpu_dai>;
+		simple-audio-card,frame-master = <&cpu_dai>;
+		simple-audio-card,mclk-fs = <256>;
+
+		simple-audio-card,widgets = "Microphone", "Microphone Jack";
+		simple-audio-card,routing =
+			    "L2", "Mic Bias",
+			    "R2", "Mic Bias",
+			    "Mic Bias", "Microphone Jack";
+
+		cpu_dai: simple-audio-card,cpu {
+			sound-dai = <&ssi3>;
+		};
+
+		codec_dai: simple-audio-card,codec {
+			sound-dai = <&tlv320aic3x>; //add TI codec for TLV320AIC3110 (tlv320aic3x)
+			clocks = <&audio_mclock>;
+		};
+	};
+
+
+	vccq_sdhi0: regulator-vccq-sdhi0 {
+		compatible = "regulator-gpio";
+
+		regulator-name = "SDHI0 VccQ";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <3300000>;
+
+		gpios = <&pinctrl RZG2L_GPIO(6, 2) GPIO_ACTIVE_HIGH>;
+
+		gpios-states = <1>;
+		states = <3300000 1
+			  1800000 0>;
+	};
+
+	vccq_sdhi1: regulator-vccq-sdhi1 {
+		compatible = "regulator-gpio";
+
+		regulator-name = "SDHI1 VccQ";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <3300000>;
+
+		gpios = <&pinctrl RZG2L_GPIO(6, 1) GPIO_ACTIVE_HIGH>;
+
+		gpios-states = <1>;
+		states = <3300000 1
+			  1800000 0>;
+	};
+	vring-ctl0@43000000 {
+		compatible = "vring_uio";
+		reg = <0x0 0x43000000 0x0 0x100000>;
+		no-map;
+	};
+
+	vring-ctl1@43100000 {
+		compatible = "vring_uio";
+		reg = <0x0 0x43100000 0x0 0x100000>;
+		no-map;
+	};
+
+	vring-shm0@43200000 {
+		compatible = "shm_uio";
+		reg = <0x0 0x43200000 0x0 0x300000>;
+		no-map;
+	};
+
+	vring-shm1@43500000 {
+		compatible = "shm_uio";
+		reg = <0x0 0x43500000 0x0 0x300000>;
+		no-map;
+	};
+
+	rsctbl@42f00000 {
+		compatible = "shm_uio";
+		reg = <0x0 0x42f00000 0x0 0x1000>;
+		no-map;
+	};
+
+	mhu-shm@42f01000 {
+		compatible = "shm_uio";
+		reg = <0x0 0x42f01000 0x0 0x1000>;
+		no-map;
+	};
+
+	mbox-uio@10400000 {
+		compatible = "mbox_uio";
+		reg = <0x0 0x10400000 0x0 0x1000>;
+		interrupt-parent = <&gic>;
+		interrupts = <GIC_SPI 75 IRQ_TYPE_LEVEL_HIGH>;
+		status = "okay";
+	};
+
+	x1_clk: x1-clock {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency = <24000000>;
+	};
+
+	clk_ext_camera: clk_ext_camera {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency = <24000000>;
+	};
+
+	ov5645_vdddo_1v8: 1p8v {
+		compatible = "regulator-fixed";
+		regulator-name = "camera_vdddo";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+		regulator-always-on;
+	};
+
+	ov5645_vdda_2v8: 2p8v {
+		compatible = "regulator-fixed";
+		regulator-name = "camera_vdda";
+		regulator-min-microvolt = <2800000>;
+		regulator-max-microvolt = <2800000>;
+		regulator-always-on;
+	};
+
+	ov5645_vddd_1v5: 1p5v {
+		compatible = "regulator-fixed";
+		regulator-name = "camera_vddd";
+		regulator-min-microvolt = <1500000>;
+		regulator-max-microvolt = <1500000>;
+		regulator-always-on;
+	};
+};
+
+&audio_clk1{
+	clock-frequency = <11289600>;
+};
+
+&audio_clk2{
+	clock-frequency = <12288000>;
+};
+
+&canfd {
+	pinctrl-0 = <&canfd0_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+	channel0 {
+		status = "okay";
+	};
+};
+
+#if (SW_DISP_EN)
+	&du {
+		pinctrl-0 = <&du_pins>;
+		pinctrl-names = "default";
+		status = "okay";
+
+		ports {
+			port@0 {
+				du_out_rgb: endpoint {
+					remote-endpoint = <&adv7513_in>;
+				};
+			};
+		};
+	};
+#else
+	&ssi0 {
+		pinctrl-0 = <&ssi0_pins>;
+		pinctrl-names = "default";
+		status = "okay";
+	};
+	
+	&canfd {
+		pinctrl-0 = <&canfd1_pins>;
+		pinctrl-names = "default";
+		status = "okay";
+		channel1 {
+			status = "okay";
+		};
+	};
+#endif
+
+&ehci0 {
+	dr_mode = "otg";
+	status = "okay";
+};
+
+&ehci1 {
+	status = "okay";
+};
+
+&eth0 {
+	pinctrl-0 = <&ether0_pins>;
+	pinctrl-names = "default";
+	phy-handle = <&phy0>;
+	phy-mode = "rgmii";
+	status = "okay";
+
+	phy0: ethernet-phy@7 {
+		reg = <7>;
+		rxc-skew-psec = <2400>;
+		txc-skew-psec = <2400>;
+		rxdv-skew-psec = <0>;
+		txdv-skew-psec = <0>;
+		rxd0-skew-psec = <0>;
+		rxd1-skew-psec = <0>;
+		rxd2-skew-psec = <0>;
+		rxd3-skew-psec = <0>;
+		txd0-skew-psec = <0>;
+		txd1-skew-psec = <0>;
+		txd2-skew-psec = <0>;
+		txd3-skew-psec = <0>;
+
+		interrupt-parent = <&pinctrl>;
+		interrupts = <RZG2L_GPIO(5, 1) IRQ_TYPE_EDGE_FALLING>;
+	};
+};
+
+&hsusb {
+	dr_mode = "otg";
+	status = "okay";
+};
+
+&i2c0 {
+	pinctrl-0 = <&i2c0_pins>;
+	pinctrl-names = "default";
+
+	status = "okay";
+
+	ov5645: camera@3c {
+		compatible = "ovti,ov5645";
+		reg = <0x3c>;
+		clock-names = "xclk";
+		clocks = <&clk_ext_camera>;
+		clock-frequency = <24000000>;
+		vdddo-supply = <&ov5645_vdddo_1v8>;
+		vdda-supply = <&ov5645_vdda_2v8>;
+		vddd-supply = <&ov5645_vddd_1v5>;
+
+		enable-gpios = <&pinctrl RZG2L_GPIO(4, 4) GPIO_ACTIVE_HIGH>; 
+
+		reset-gpios = <&pinctrl RZG2L_GPIO(0, 1) GPIO_ACTIVE_LOW>;
+
+		port {
+			ov5645_to_csi: endpoint {
+				clock-lanes = <0>;
+				data-lanes = <1 2>;
+				remote-endpoint = <&csi2_in>;
+			};
+		};
+	};
+};
+
+&i2c1 {
+	pinctrl-0 = <&i2c1_pins>;
+	pinctrl-names = "default";
+
+	status = "okay";
+
+	adv7513: adv7513@39 {
+		compatible = "adi,adv7513";
+		reg = <0x39>;
+
+		adi,input-depth = <8>;
+		adi,input-colorspace = "rgb";
+		adi,input-clock = "1x";
+
+		avdd-supply = <&reg_1p8v>;
+		dvdd-supply = <&reg_1p8v>;
+		pvdd-supply = <&reg_1p8v>;
+		dvdd-3v-supply = <&reg_3p3v>;
+		bgvdd-supply = <&reg_1p8v>;
+
+		ports {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			port@0 {
+				reg = <0>;
+
+				adv7513_in: endpoint {
+					remote-endpoint = <&du_out_rgb>;
+				};
+			};
+
+			port@1 {
+				reg = <1>;
+
+				adv7513_out: endpoint {
+					remote-endpoint = <&hdmi_con_out>;
+				};
+			};
+		};
+	};
+	
+
+	tlv320aic3x: tlv320aic3x@1b {
+		compatible = "ti,tlv320aic3x";
+		reg = <0x1b>;
+		//reset-gpios = <&gpio1 17 GPIO_ACTIVE_LOW>;  check reset
+	};
+};
+
+&i2c2 {
+    pinctrl-0 = <&i2c2_pins>;
+    pinctrl-names = "default";
+    status = "okay";
+};
+
+
+&i2c3 {
+    pinctrl-0 = <&i2c3_pins>;
+    pinctrl-names = "default";
+    status = "okay";
+};
+
+&ssi3 {
+	    pinctrl-0 = <&ssi3_pins>;
+	    pinctrl-names = "default";
+	    status = "okay";
+};
+
+&usb2_phy0 {
+	pinctrl-0 = <&usb0_pins>;
+	pinctrl-names = "default";
+
+	vbus-supply = <&vbus0_usb2>;
+	status = "okay";
+};
+
+&usb2_phy1 {
+	pinctrl-0 = <&usb1_pins>;
+	pinctrl-names = "default";
+
+	status = "okay";
+};
+
+&spi0 {
+	    pinctrl-0 = <&spi0_pins>;
+	    pinctrl-names = "default";
+	    status = "okay";
+};
+
+&spi2 {
+	    pinctrl-0 = <&spi2_pins>;
+	    pinctrl-names = "default";
+	    status = "okay";
+};
+
+&scif0 {
+	pinctrl-0 = <&scif0_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+};
+	
+
+&scif1 {
+    pinctrl-0 = <&scif1_pins>;
+    pinctrl-names = "default";
+    status = "okay";
+};
+
+&scif2 {
+    pinctrl-0 = <&scif2_pins>;
+    pinctrl-names = "default";
+    status = "okay";
+};
+
+
+&scif4 {
+    pinctrl-0 = <&scif4_pins>;
+    pinctrl-names = "default";
+    status = "okay";
+};
+
+//Comments:sci0 : M3 and M6 pins are used but Name is not verified with r9...It is serial.
+
+&sci0 {
+    pinctrl-0 = <&sci0_pins>;
+    pinctrl-names = "default";
+    status = "okay";
+};
+	
+&phyrst {
+	status = "okay";
+};
+
+&ohci0 {
+	dr_mode = "otg";
+	status = "okay";
+};
+
+&ohci1 {
+	status = "okay";
+};
+
+&pinctrl {
+	pinctrl-0 = <&sound_clk_pins>;
+	pinctrl-names = "default";
+
+
+	i2c2_pins: i2c2 {
+               pinmux = <RZG2L_PORT_PINMUX(0, 3, 4)>, /* RIIC2_SCL */
+                         <RZG2L_PORT_PINMUX(0, 2, 4)>; /*RIIC2_SDA*/
+	};
+	
+	i2c3_pins: i2c3 {
+               pinmux = <RZG2L_PORT_PINMUX(0, 1, 3)>, /* RIIC3_SCL */
+                         <RZG2L_PORT_PINMUX(0, 0, 3)>; /*RIIC3_SDA*/
+	};
+
+	canfd0_pins: can0 {
+		pinmux = <RZG2L_PORT_PINMUX(1, 1, 3)>, /* TX */
+			 <RZG2L_PORT_PINMUX(1, 2, 3)>; /* RX */
+	};
+
+	ether0_pins: eth0 {
+		pinmux = <RZG2L_PORT_PINMUX(4, 5, 1)>, /* ET0_LINKSTA */
+			 <RZG2L_PORT_PINMUX(4, 3, 1)>, /* ET0_MDC */
+			 <RZG2L_PORT_PINMUX(4, 4, 1)>, /* ET0_MDIO */
+			 <RZG2L_PORT_PINMUX(1, 0, 1)>, /* ET0_TXC */
+			 <RZG2L_PORT_PINMUX(1, 1, 1)>, /* ET0_TX_CTL */
+			 <RZG2L_PORT_PINMUX(1, 2, 1)>, /* ET0_TXD0 */
+			 <RZG2L_PORT_PINMUX(1, 3, 1)>, /* ET0_TXD1 */
+			 <RZG2L_PORT_PINMUX(1, 4, 1)>, /* ET0_TXD2 */
+			 <RZG2L_PORT_PINMUX(2, 0, 1)>, /* ET0_TXD3 */
+			 <RZG2L_PORT_PINMUX(3, 0, 1)>, /* ET0_RXC */
+			 <RZG2L_PORT_PINMUX(3, 1, 1)>, /* ET0_RX_CTL */
+			 <RZG2L_PORT_PINMUX(3, 2, 1)>, /* ET0_RXD0 */
+			 <RZG2L_PORT_PINMUX(3, 3, 1)>, /* ET0_RXD1 */
+			 <RZG2L_PORT_PINMUX(4, 0, 1)>, /* ET0_RXD2 */
+			 <RZG2L_PORT_PINMUX(4, 1, 1)>; /* ET0_RXD3 */
+	};
+
+	i2c0_pins: i2c0 {
+		pins = "RIIC0_SDA", "RIIC0_SCL";
+		input-enable;
+	};
+
+	i2c1_pins: i2c1 {
+		pins = "RIIC1_SDA", "RIIC1_SCL";
+		input-enable;
+	};
+
+
+
+	sd1_pwr_en {
+		gpio-hog;
+		gpios = <RZG2L_GPIO(0, 3) GPIO_ACTIVE_HIGH>;
+		output-high;
+		line-name = "sd1_pwr_en";
+	};
+
+
+/* Support pinctrl for uSD function of SDHI0*/
+//For SD Card
+	sdhi0_pins: sd0 {
+		sd0_data {
+			pins =  "SD0_DATA0", "SD0_DATA1", "SD0_DATA2",
+				"SD0_DATA3";
+			power-source  = <3300>;
+		};
+
+		sd0_ctrl {
+			pins = "SD0_CLK", "SD0_CMD";
+			power-source = <3300>;
+		};
+	};
+
+	sdhi0_pins_uhs: sd0_uhs {
+		sd0_data_uhs {
+			pins =	"SD0_DATA0", "SD0_DATA1", "SD0_DATA2",
+				"SD0_DATA3";
+			power-source  = <1800>;
+		};
+
+		sd0_ctrl_uhs {
+			pins = "SD0_CLK", "SD0_CMD";
+			power-source  = <1800>;
+		};
+	};
+	
+// For MMC
+	sdhi1_pins: sd1 {
+		sd1_data {
+			pins =	"SD1_DATA0", "SD1_DATA1",
+				"SD1_DATA2", "SD1_DATA3";
+			power-source  = <1800>;
+		};
+
+		sd1_ctrl {
+			pins = "SD1_CLK", "SD1_CMD";
+			power-source  = <1800>;
+		};
+	};
+	
+	
+#if (SW_DISP_EN)
+
+	du_pins: du {
+		pinmux = <RZG2L_PORT_PINMUX(11, 2, 6)>,
+			 <RZG2L_PORT_PINMUX(13, 1, 6)>,
+			 <RZG2L_PORT_PINMUX(13, 0, 6)>,
+			 <RZG2L_PORT_PINMUX(13, 4, 6)>,
+			 <RZG2L_PORT_PINMUX(13, 3, 6)>,
+			 <RZG2L_PORT_PINMUX(12, 1, 6)>,
+			 <RZG2L_PORT_PINMUX(13, 2, 6)>,
+			 <RZG2L_PORT_PINMUX(14, 0, 6)>,
+			 <RZG2L_PORT_PINMUX(14, 2, 6)>,
+			 <RZG2L_PORT_PINMUX(14, 1, 6)>,
+			 <RZG2L_PORT_PINMUX(16, 0, 6)>,
+			 <RZG2L_PORT_PINMUX(15, 0, 6)>,
+			 <RZG2L_PORT_PINMUX(16, 1, 6)>,
+			 <RZG2L_PORT_PINMUX(15, 1, 6)>,
+			 <RZG2L_PORT_PINMUX(15, 3, 6)>,
+			 <RZG2L_PORT_PINMUX(18, 0, 6)>,
+			 <RZG2L_PORT_PINMUX(15, 2, 6)>,
+			 <RZG2L_PORT_PINMUX(17, 0, 6)>,
+			 <RZG2L_PORT_PINMUX(17, 2, 6)>,
+			 <RZG2L_PORT_PINMUX(17, 1, 6)>,
+			 <RZG2L_PORT_PINMUX(18, 1, 6)>,
+			 <RZG2L_PORT_PINMUX(18, 2, 6)>,
+			 <RZG2L_PORT_PINMUX(17, 3, 6)>,
+			 <RZG2L_PORT_PINMUX(18, 3, 6)>,
+			 <RZG2L_PORT_PINMUX(11, 0, 6)>, /* HSYNC */
+			 <RZG2L_PORT_PINMUX(12, 0, 6)>, /* VSYNC */
+			 <RZG2L_PORT_PINMUX(11, 1, 6)>, /* DE */
+			 <RZG2L_PORT_PINMUX(11, 3, 6)>; /* CLK */
+	};
+#else
+	ssi0_pins: ssi0 {
+		pinmux = <RZG2L_PORT_PINMUX(11, 0, 1)>, /* SSI0_BCK */
+			 <RZG2L_PORT_PINMUX(11, 1, 1)>, /* SSI0_RCK */
+			 <RZG2L_PORT_PINMUX(11, 2, 1)>, /* SSI0_TXD */
+			 <RZG2L_PORT_PINMUX(11, 3, 1)>; /*SSI0_RXD*/
+
+	};
+	
+	canfd1_pins: can1 {
+		pinmux = <RZG2L_PORT_PINMUX(2, 0, 3)>, /* TX */
+			 <RZG2L_PORT_PINMUX(2, 1, 3)>; /* RX */
+	};
+
+#endif
+	
+	ssi3_pins: ssi3 {
+		       pinmux = <RZG2L_PORT_PINMUX(10, 1, 3)>, /* SSI3_BCK */
+		                 <RZG2L_PORT_PINMUX(10, 2, 3)>, /*SSI3_RCK*/
+		                 <RZG2L_PORT_PINMUX(10, 3, 3)>, /*SSI3_TXD*/
+		                 <RZG2L_PORT_PINMUX(10, 4, 3)>; /*SSI3_RXD*/
+	};
+
+
+	sound_clk_pins: sound_clk {
+		pins = "AUDIO_CLK1", "AUDIO_CLK2";
+		input-enable;
+	};
+
+
+	usb0_pins: usb0 {
+		pinmux = <RZG2L_PORT_PINMUX(5, 0, 1)>, /* VBUS */
+			 <RZG2L_PORT_PINMUX(5, 2, 1)>, /* OVC */
+			 <RZG2L_PORT_PINMUX(5, 3, 1)>; /* OTG_ID */
+	};
+
+	usb1_pins: usb1 {
+		pinmux = <RZG2L_PORT_PINMUX(5, 4, 5)>, /* VBUS */
+			 <RZG2L_PORT_PINMUX(6, 0, 1)>; /* OVC */
+	};
+
+
+
+	spi0_pins: rspi0 {
+		       pinmux = <RZG2L_PORT_PINMUX(1, 0, 2)>, /* RSPI0_CK */
+		                 <RZG2L_PORT_PINMUX(1, 1, 2)>, /*RSPI0_MOSI*/
+		                 <RZG2L_PORT_PINMUX(1, 2, 2)>, /*RSPI0_MISO*/
+		                 <RZG2L_PORT_PINMUX(1, 3, 2)>; /*RSPI0_SSL*/
+	};
+
+
+	spi2_pins: rspi2 {
+		       pinmux = <RZG2L_PORT_PINMUX(6, 0, 2)>, /* RSPI2_CK */
+		                 <RZG2L_PORT_PINMUX(6, 1, 2)>, /*RSPI2_MOSI*/
+		                 <RZG2L_PORT_PINMUX(6, 2, 2)>, /*RSPI2_MISO*/
+		                 <RZG2L_PORT_PINMUX(6, 3, 2)>; /*RSPI2_SSL*/
+	};
+	
+	scif0_pins: scif0 {
+			pinmux = <RZG2L_PORT_PINMUX(6, 4, 6)>, /* TxD */
+				 <RZG2L_PORT_PINMUX(6, 3, 6)>; /* RxD */
+	};
+
+
+	scif1_pins: scif1 {
+		       pinmux = <RZG2L_PORT_PINMUX(8, 0, 3)>, /* SCIF1_RXD */
+		                 <RZG2L_PORT_PINMUX(8, 1, 3)>,/*SCIF1_TXD*/
+		                 <RZG2L_PORT_PINMUX(8, 2, 3)>, /*SCIF1_CTS*/
+		                 <RZG2L_PORT_PINMUX(8, 3, 3)>; /*SCIF1_RTS*/
+	};
+
+
+
+	scif2_pins: scif2 {
+		       pinmux = <RZG2L_PORT_PINMUX(5, 0, 2)>, /* SCIF2_TXD */
+		                 <RZG2L_PORT_PINMUX(5, 1, 2)>, /*SCIF2_RXD*/
+		                 <RZG2L_PORT_PINMUX(5, 2, 2)>, /*SCIF2_SCK*/
+		                 <RZG2L_PORT_PINMUX(5, 3, 2)>, /*SCIF2_CTS*/
+		                 <RZG2L_PORT_PINMUX(5, 4, 2)>; /*SCIF2_RTS*/
+	};
+
+
+	scif4_pins: scif4 {
+		       pinmux = <RZG2L_PORT_PINMUX(9, 0, 6)>, /* SCIF4_SCK */
+		                 <RZG2L_PORT_PINMUX(9, 1, 6)>, /*SCIF4_RXD*/
+		                 <RZG2L_PORT_PINMUX(9, 2, 6)>; /*SCIF4_TXD*/
+	};
+	
+	sci0_pins: sci0 {
+               pinmux = <RZG2L_PORT_PINMUX(2, 3, 5)>, /*SCI0_TXD*/
+                         <RZG2L_PORT_PINMUX(2, 2, 5)>; /* SCI0_RXD*/ 
+	};
+
+};
+
+&sbc {
+	status = "okay";
+
+	flash@0 {
+		compatible = "micron,mt25qu512a", "jedec,spi-nor";
+		reg = <0>;
+		m25p,fast-read;
+		spi-max-frequency = <50000000>;
+		spi-tx-bus-width = <1>;
+		spi-rx-bus-width = <4>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@000000 {
+				label = "bl2";
+				reg = <0x00000000 0x0001D200>;
+				read-only;
+			};
+			partition@01D200 {
+				label = "fip";
+				reg = <0x0001D200 0x001C2E00>;
+				read-only;
+			};
+			partition@1E0000 {
+				label = "env";
+				reg = <0x001E0000 0x00020000>;
+				read-only;
+			};
+			partition@200000 {
+				label = "test-area";
+				reg = <0x00200000 0x00E00000>;
+			};
+		};
+	};
+};
+
+
+
+
+//SD Card
+&sdhi0 {
+	pinctrl-0 = <&sdhi0_pins>;
+	pinctrl-1 = <&sdhi0_pins>;
+	pinctrl-names = "default", "state_uhs";
+
+	vmmc-supply = <&reg_3p3v>;
+	vqmmc-supply = <&reg_1p8v>;
+	bus-width = <8>;
+	mmc-hs200-1_8v;
+	non-removable;
+	fixed-emmc-driver-type = <1>;
+	status = "okay";
+};
+
+&sdhi0 {
+	pinctrl-0 = <&sdhi0_pins>;
+	pinctrl-1 = <&sdhi0_pins_uhs>;
+	pinctrl-names = "default", "state_uhs";
+
+	vmmc-supply = <&reg_3p3v>;
+	vqmmc-supply = <&vccq_sdhi0>;
+	cd-gpios = <&pinctrl RZG2L_GPIO(0, 0) GPIO_ACTIVE_LOW>;
+	bus-width = <4>;
+	sd-uhs-sdr50;
+	sd-uhs-sdr104;
+	status = "okay";
+};
+
+//MMC
+&sdhi1 {
+	pinctrl-0 = <&sdhi1_pins>;
+	pinctrl-1 = <&sdhi1_pins>;
+	pinctrl-names = "default", "state_uhs";
+
+	vmmc-supply = <&reg_3p3v>;
+	vqmmc-supply = <&reg_1p8v>;
+	bus-width = <4>;
+	mmc-hs200-1_8v;
+	non-removable;
+	fixed-emmc-driver-type = <1>;
+	status = "okay";
+};
+
+
+&extal_clk {
+	clock-frequency = <24000000>;
+};
+
+&cru {
+	status = "okay";
+};
+
+&csi2 {
+	status = "okay";
+
+	ports {
+		port {
+			csi2_in: endpoint {
+				clock-lanes = <0>;
+				data-lanes = <1 2>;
+				remote-endpoint = <&ov5645_to_csi>;
+			};
+		};
+	};
+};
+
+&wdt0 {
+	status = "okay";
+	timeout-sec = <60>;
+};
+
+&wdt2 {
+	status = "okay";
+	timeout-sec = <60>;
+};
+
+&ostm1 {
+	status = "okay";
+};
+
+&ostm2 {
+	status = "disabled";
+};
+
+&mtu3 {
+	status = "okay";
+};
+
-- 
2.25.1

